drivers/veteran_confirmation/app/models/veteran_confirmation/checker.rb

```diff
@@ -0,0 +1,49 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module VeteranConfirmation
+  class Checker
+    def check(clients)
+      client_id = 0
+      {}.tap do |results|
+        last_client_id = clients.pluck(:id).max
+        clients.find_in_batches(batch_size: 60) do |batch|
+          limiter = RateLimiter.new
+          batch.each do |client|
+            client_id = client.id
+            results[client_id] = status(client)
+          end
+          # Avoid rate limiting at the end of the run
+          limiter.drain if client_id != last_client_id
+        end
+      end
+    end
+
+    private def status(client)
+      credentials = Credential.first
```

Comment:
`credentials` should be memoized, this is potentially loading that record for every client.